packages/iptables/data_stream/log/elasticsearch/ingest_pipeline/default.yml

```
@@ -248,7 +248,49 @@ processors:
             }
           }
         }
+  - script:
+      description: Convert bit fields to numbers.
+      lang: painless
+      if: >-
+        ctx?.network?.iana_number == null && ctx?.network?.transport != null
+      source: >-
+        def transport = ctx?.network?.transport;
+        if (transport == '1') {
+          ctx.network.iana_number = '1';
+        } else if (transport == '2') {
+          ctx.network.iana_number = '2';
+        } else if (transport == '6') {
+          ctx.network.iana_number = '6';
+        } else if (transport == '17') {
+          ctx.network.iana_number = '17';
+        } else if (transport == '47') {
+          ctx.network.iana_number = '47';
+        } else if (transport == '58') {
+          ctx.network.iana_number = '58';
+        } else if (transport == '88') {
+          ctx.network.iana_number = '88';
+        } else if (transport == '89') {
+          ctx.network.iana_number = '89';
+        } else if (transport == '103') {
+          ctx.network.iana_number = '103';
+        } else if (transport == '132') {
+          ctx.network.iana_number = '132';
+        }
   - community_id:
+      if: >-
+        ctx?.network?.transport == null || 
+        ctx?.network?.iana_number != null ||
+        ctx?.network?.transport == 'icmp' || 
+        ctx?.network?.transport == 'igmp' || 
+        ctx?.network?.transport == 'tcp' || 
+        ctx?.network?.transport == 'udp' || 
+        ctx?.network?.transport == 'gre' || 
+        ctx?.network?.transport == 'ipv6-icmp' || 
+        ctx?.network?.transport == 'icmpv6' || 
+        ctx?.network?.transport == 'eigrp' || 
+        ctx?.network?.transport == 'ospf' || 
+        ctx?.network?.transport == 'pim' || 
+        ctx?.network?.transport == 'sctp'
```

An alternative would be to add `ignore_failure: true`. Simpler, but cast a wider net and may mask additional problems.

---

hmm more convenient but might cause silent error that we actually care for 🤔

---

ok I have done it this way. Please feel free to unresolve if you spot anything else